python_modules/dagster/dagster/_core/definitions/declarative_scheduling/operands/scheduled_since_condition.py

```diff
@@ -0,0 +1,142 @@
+import datetime
+from typing import Sequence, Tuple, cast
+
+from dagster._core.definitions.declarative_scheduling.serialized_objects import (
+    AssetSubsetWithMetadata,
+)
+from dagster._core.definitions.metadata.metadata_value import MetadataValue
+from dagster._serdes.serdes import whitelist_for_serdes
+
+from ..scheduling_condition import SchedulingCondition, SchedulingResult
+from ..scheduling_context import SchedulingContext
+
+_REQUEST_TIMESTAMP_METADATA_KEY = "request_timestamp"
+
+
+@whitelist_for_serdes
+class ScheduledSinceCondition(SchedulingCondition):
+    """SchedulingCondition which is true if the asset has been requested for materialization via
+    the declarative scheduling system within the given time window.
+
+    Will only detect requests which have been made since this condition was added to the asset.
+    """
+
+    lookback_days_seconds_microseconds: Tuple[int, int, int]
```

Comment:
I think we should encapsulate serialization of timedelta in a lightweight class or helper function. If nothing else this centralizes and documents our policy decision of how we choose to serialize timedeltas and why.

Reply:
Added!